src/ast.mli

```diff
@@ -76,18 +70,37 @@ and stmt =
   | Break
   | Continue
 
-type func = {
-  name : string;
-  return_t : types;
-  formals : formal list;
-  body : stmt list;
+
+type message = {
+    name: string;
+    formals: formal list;
+}
+
+type pattern = {
+  message_id: string;
+  message_formals: formals list;
+  stmts: stmt list;
 }
 
 type actor = {
-  fields : field list;
-  name : string;
-  formals : formal list;
-  receive: message_type list;
+  name: string;
+  formals: formal list;
+  body: stmt list;
+  functions: func list;
+  receive: pattern list;
+}
+
+type func = {
+  name: string;
+  return_t: types;
+  formals: formal list;
+  body: stmt list;
+}
+
+type lambda = {
+  formals: formal list;
+  return_t: types;
+  return_expr: expr;
```

Comment:
I wonder why `lambda` has `return_expr: expr;` as opposed to `body: stmt list;` as in `func`.
Isn't lambda basically a nameless function?